avaframe/in3Utils/geoTrans.py

```diff
@@ -288,35 +291,85 @@ def remeshDEM(cfg, dem):
+            log.debug('remeshed DEM found')
+        else:
+            log.info('Remeshing the input DEM (of cell size %.4g m) to a cell size of %.4g m' % (cszDEM, cszNew))
+            x, y, xNew, yNew, diffExtentX, diffExtentY = getMeshXY(dem, cellSizeNew=cszNew)
+            xGrid, yGrid = np.meshgrid(x, y)
+            xGrid = xGrid.flatten()
+            yGrid = yGrid.flatten()
+            z = dem['rasterData']
+            zCopy = np.copy(z)
+            zCopy = zCopy.flatten()
+            mask = np.where(~np.isnan(zCopy))
+            xGrid = xGrid[mask]
+            yGrid = yGrid[mask]
+            z = zCopy[mask]
+
+            headerRemeshed = {}
+            headerRemeshed['cellsize'] = cszNew
+            headerRemeshed['ncols'] = len(xNew)
+            headerRemeshed['nrows'] = len(yNew)
+            headerRemeshed['xllcenter'] = headerDEM['xllcenter']
+            headerRemeshed['yllcenter'] = headerDEM['yllcenter']
+            headerRemeshed['noDataValue'] = headerDEM['noDataValue']
+
+            remeshedDem = {'header': headerRemeshed}
+            xNewGrid, yNewGrid = np.meshgrid(xNew, yNew)
+            zNew = sp.interpolate.griddata((xGrid, yGrid), z, (xNewGrid, yNewGrid), method='cubic',
+                                           fill_value=headerDEM['noDataValue'])
+            log.info('Remeshed data extent difference x: %f and y %f' % (diffExtentX, diffExtentY))
+            remeshedDem['rasterData'] = zNew
+
+            # save remeshed DEM
+            pathToDem = pathlib.Path(cfg['avalancheDir'], 'Inputs', 'DEMremeshed')
+            fU.makeADir(pathToDem)
+            outFile = pathToDem / ('remeshedDEM%.2f.asc' % remeshedDem['header']['cellsize'])
+            IOf.writeResultToAsc(remeshedDem['header'], remeshedDem['rasterData'], outFile, flip=True)
+
+    return remeshedDem
+
+
+def searchRemeshedDEM(cfg):
+    """ search if remeshed DEM already available
+
+        Parameters
+        -----------
+        cfg: configparser object
+            configuration settings: avaDir, meshCellSize, meshCellSizeThreshold
+
+        Returns
+        --------
+        remshedDEM: dict
+            dictionary of remeshed DEM if not found empty dict
+        DEMFound: bool
+            flag if dem is found
+    """
+
+    # load input parameters
+    cellSize = cfg.getfloat('meshCellSize')
+    cellSizeThreshold = cfg.getfloat('meshCellSizeThreshold')
+
+    # path to remeshed DEM folder
+    pathToDem = pathlib.Path(cfg['avalancheDir'], 'Inputs', 'DEMremeshed')
+    fU.makeADir(pathToDem)
+
+    # look for dems
+    demFiles = list(pathToDem.glob('*.asc'))
+    DEMFound = False
+    dem = {}
+    for demF in demFiles:
+        demDict = IOf.readRaster(demF)
+        if abs(cellSize - demDict['header']['cellsize']) < cellSizeThreshold:
+            log.info('Remeshed DEM found: %s cellSize: %.5f' % (demF.name, demDict['header']['cellsize']))
+            DEMFound = True
+            dem = demDict
+            continue
```

Comment:
suggestion. Maybe we could make kind of a csv file with the info about all remeshed dem. Then you could look at this dataframe to choose the good dem?

Reply:
how should we do this? every time a remeshedDEM is stored we open the csv and append the info? but then it is not allowed to store a DEM from outside there, and if we read the DEM first to create the csv. it takes more time

Reply:
I´m not a fan of creating a csv! We are not talking about a common occurance with tons of files. Reading a dem and getting the header is not huge expense...